design-documents/node-static-content-deploy.md

```diff
@@ -0,0 +1,95 @@
+# Modernize Static Content Deployment
+
+**Note: This is a proposal for a possible improvement in the future. Even if we reach consensus, there are no guarantees yet that this would be implemented. Exploratory for the moment**
+
+- [Static Content Deployment](#static-content-deployment)
+  * [Docs](#docs)
+  * [Use Cases](#use-cases)
+  * [What does it do?](#what-does-it-do-)
+  * [Problems with Static Content Deployment](#problems-with-static-content-deployment)
+  * [Why rewrite Static Content Deployment?](#why-rewrite-static-content-deployment-)
+    + [Performance](#performance)
+    + [Access to Modern Tooling](#access-to-modern-tooling)
+    + [Accessible for Front-End Devs](#accessible-for-front-end-devs)
+    + [What opportunities open up with a re-write of SCD using node.js?](#what-opportunities-open-up-with-a-re-write-of-scd-using-nodejs-)
+  * [How do you re-write Static Content Deployment?](#how-do-you-re-write-static-content-deployment-)
+  
+## Docs
+- [Deploying Static View Files](https://devdocs.magento.com/guides/v2.3/config-guide/cli/config-cli-subcommands-static-view.html)
+- [Strategies](https://devdocs.magento.com/guides/v2.3/config-guide/cli/config-cli-subcommands-static-deploy-strategies.html)
+
+## Use Cases
+1. Pre-processing and flattening of static assets in local development (i.e. less)
+1. Pre-processing and flattening of static assets for production deployment
+
+## What does it do?
+
+- [Theme/module/i18n/"lib/web" file inheritance flattening](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/themes/theme-inherit.html)
+- Locale splitting
+    - Each locale gets its own static assets under a theme dir
+- Less compilation
+- HTML minification
+- JS minification
+- JS concatenation
+- `js-translation.json` generation
+- `requirejs-config.js` generation
+- Asset signing/versioning
+
+## Problems with Static Content Deployment
+
+There are a number of issues with the current implementation of static content deployment.
+
+1. It's prohibitively slow, sometimes taking 10+ minutes just to copy and process files
```

Comment:
It always depends on the number of files to deploy but for Magento vanilla installation (CE + EE) it takes 13sec on my laptop.

Reply:
Right, you're on the happy path it was designed for, which is part of the problem: those speeds are only available on stores with a limited number of files and only 1 locale.

If it takes several minutes for multiple locales, and the majority of our user base in Europe is using multiple locales, then the speed of a basic store isn't super relevant.